infrastructure/persistence/file.go

```
@@ -159,34 +163,113 @@
+		return wrapXZReader(reader, closer)
+	}
+	if file.IsZSTD() {
+		return wrapZstdReader(reader, closer)
+	}
+	return reader, closer, nil
+}
+
+func wrapGZReader(reader io.Reader, closer func() error) (io.Reader, func() error, error) {
+	gzReader, err := gzip.NewReader(reader)
+	if err != nil {
+		closer()
```

golangci-lint (1.64.8)

Error return value is not checked

(errcheck)